odl/trafos/convolution.py

```diff
@@ -0,0 +1,849 @@
+    def integral(self):
+        """``True`` if the conv. integral is approximated, else ``False``"""
+        return self.__integral
+
+    @property
+    def ft_op(self):
+        """Fourier transform operator used in this convolution."""
+        return self.__ft_op
+
+    @property
+    def ft_impl(self):
+        """Implementation of the Fourier transform in this convolution."""
+        return self.ft_op.impl
+
+    @property
+    def axes(self):
+        """Axes along which the convolution is taken."""
+        return self.ft_op.axes
+
+    @property
+    def kernel_ft_op(self):
+        """The Fourier transform operator for the kernel.
+
+        Its domain may differ from that of `ft_op`, but the ranges of both
+        transforms are the same.
+        """
+        # TODO: Update docstring when broadcasting convolution is there
+        return self.__kernel_ft_op
+
+    def kernel_ft(self, out=None):
+        """Return a the FT of this convolution operator's kernel.
+
+        Parameters
+        ----------
+        out : element of ``kernel_ft_op.range``, optional
+            Write the output to this data storage.
+
+        Returns
+        -------
+        kernel_ft : element of ``kernel_ft_op.range``
+            FT of the convolution kernel. If ``out`` was given, this
+            object is a reference to it.
+        """
+        if out is None:
+            out = self.kernel_ft_op.range.element()
+
+        out.lincomb(1 / self.__kernel_scaling, self.__scaled_kernel_ft)
+        return out
+
+    def kernel(self, out=None):
+        """Return the real-space kernel of this convolution operator.
+
+        Parameters
+        ----------
+        out : element of ``kernel_ft_op.domain``, optional
+            Write the output to this data storage.
+
+        Returns
+        -------
+        kernel : element of ``kernel_ft_op.domain``
+            The Convolution kernel. If ``out`` was given, this
+            object is a reference to it.
+        """
+        if out is None:
+            out = self.kernel_ft_op.domain.element()
+
+        self.kernel_ft_op.inverse(self.__scaled_kernel_ft, out=out)
+        out /= self.__kernel_scaling
+        return out
+
+    def _call(self, x, out, **kwargs):
+        """Implement ``self(x, out[, **kwargs])``.
+
+        Keyword arguments are passed on to the transform.
+        """
+        x_trafo = self.ft_op(x, out=self.__ft_range_tmp, **kwargs)
+        # TODO: use broadcasting
+        x_trafo *= self.__scaled_kernel_ft
+        self.ft_op.inverse(x_trafo, out=out, **kwargs)
+
+    @property
+    def adjoint(self):
+        """Adjoint operator given by the reversed kernel.
+
+        This can be expressed as multiplication with the complex
+        conjugate kernel in Fourier space, which is used in this
+        implementation.
+        """
+        return self.__adjoint
+
+    @property
+    def inverse(self):
+        """Poor man's deconvolution using the reciprocal kernel FT."""
```

Comment:
I thought we had an agreement to not do poor mans inverses? You explicitly argued against this for the FBP as inverse, while I was in favor.

Just stating this for thoughts, personally I'm in favor of this.